### PR TITLE
Fix Timezone Test with Morocco on Windows

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/corefx/issues/37389

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @krwq @stephentoub @eanova

File: src/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 1]
```diff
@@ -2221,12 +2221,23 @@ public static void TimeZoneInfo_DisplayNameStartsWithOffset()
...
                     Assert.Matches(@"^\(UTC(\+|-)[0-9]{2}:[0-9]{2}\) \S.*", tzi.DisplayName);
-                    
+
                     // see https://github.com/dotnet/corefx/pull/33204#issuecomment-438782500
                     if (PlatformDetection.IsNotWindowsNanoServer && !PlatformDetection.IsWindows7)
                     {
                         string offset = Regex.Match(tzi.DisplayName, @"(-|)[0-9]{2}:[0-9]{2}").Value;
-                        Assert.True(tzi.BaseUtcOffset == TimeSpan.Parse(offset), $"{offset} != {tzi.BaseUtcOffset}, dn:{tzi.DisplayName}, sn:{tzi.DisplayName}");
+                        TimeSpan ts = TimeSpan.Parse(offset);
+                        if (tzi.BaseUtcOffset != ts && tzi.Id.IndexOf("Morocco", StringComparison.Ordinal) >= 0)
```
Comment (krwq) [inline src/System.Runtime/tests/System/TimeZoneInfoTests.cs]: is there some bug for OS to fix this?

**Comment by tarekgh** [comment]
please hold and don't merge this. I believe the test should be modified as there is wrong assumption in this test.

**Comment by tarekgh** [comment]
Looks the failed test is unrelated and tracked by https://github.com/dotnet/corefx/issues/37370

https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F37419~2Fmerge/test~2Ffunctional~2Fcli~2F/20190503.31/workItem/System.Net.Ping.Functional.Tests/analysis/xunit/System.Net.NetworkInformation.Tests.PingTest~2FSendPingAsyncWithHostAndTtlAndFragmentPingOptions(fragment:%20False)

System.Net.NetworkInformation.Tests.PingTest

```
OSX.1012.Amd64.Open-x64-Debug
Get Repro environment
Unhandled Exception of Type Xunit.Sdk.EqualException
Message :
Assert.Equal() Failure
Expected: Success
Actual:   TimedOut
Stack Trace :
   at System.Net.NetworkInformation.Tests.PingTest.PingResultValidator(PingReply pingReply, IPAddress[] localIpAddresses) in /Users/vsts/agent/2.150.3/work/1/s/src/System.Net.Ping/tests/FunctionalTests/PingTest.cs:line 61
   at System.Net.NetworkInformation.Tests.PingTest.<>c__DisplayClass46_0.<SendPingAsyncWithHostAndTtlAndFragmentPingOptions>b__1(PingReply pingReply) in /Users/vsts/agent/2.150.3/work/1/s/src/System.Net.Ping/tests/FunctionalTests/PingTest.cs:line 806
   at System.Net.NetworkInformation.Tests.PingTest.SendPingAsync(Func`2 sendPing, Action`1 pingResultValidator) in /Users/vsts/agent/2.150.3/work/1/s/src/System.Net.Ping/tests/FunctionalTests/PingTest.cs:line 776
   at System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithHostAndTtlAndFragmentPingOptions(Boolean fragment) in /Users/vsts/agent/2.150.3/work/1/s/src/System.Net.Ping/tests/FunctionalTests/PingTest.cs:line 802
--- End of stack trace from previous location where exception was thrown ---
```

**Pull request merged**